library/Booster/Definition/Attributes/Base.hs

```diff
@@ -185,13 +188,20 @@ pattern CanBeEvaluated = Flag True
 pattern CannotBeEvaluated = Flag False
 {-# COMPLETE CanBeEvaluated, CannotBeEvaluated #-}
 
+data SMTType
+    = SMTLib ByteString
```

Comment:
According to https://github.com/runtimeverification/haskell-backend/blob/b2bade8ce62e5f3add32d87e98244ceb31790f9f/docs/attributes.md?plain=1#L240 we could have `smtlib{}("(mod (^ #1 #2) #3)")`. That doesn't seem correct tho.

Reply:
I have to guess, but it may be that the `smt-hook` attribute has been separated from `smtlib` after this documentation had been written. I've grepped through `domains.md` and `evm-semantics`, and I have not found any `smtlib` attributes containing s-expressions.

Reply:
Interesting.... I have no idea how one would declare the functions in such an s-expression given in an `smtlib` attribute :thinking: Of course I can dig up the old backend code to find out what exactly it does but I doubt it would work.